src/snowflake/snowpark/dataframe_reader.py

```diff
@@ -1016,11 +1022,12 @@ def dbapi(
         lower_bound: Optional[Union[str, int]] = None,
         upper_bound: Optional[Union[str, int]] = None,
         num_partitions: Optional[int] = None,
-        predicates: Optional[List[str]] = None,
         *,
         max_workers: Optional[int] = None,
+        query_timeout: Optional[int] = 0,
     ) -> DataFrame:
         conn = create_connection()
+        conn.timeout = query_timeout
```

Comment:
I'm curious do you know what's the scope of this timeout under the context of pyodbc.
does it affect other timeout like establishing connection

Reply:
add comment, this is specific to pyodbc, we may need to handle other driver differently